#!/usr/bin/env python
PACKAGE = "path_smoothing"

from dynamic_reconfigure.parameter_generator_catkin import*

gen = ParameterGenerator()

gen.add("delta_width", double_t, 0, "Track width increment used to determine smoothing area with regard to costmap.", 0.2, 0.01, 0.5)
gen.add("num_of_iteratons", int_t, 0, "Number of path_smoothing algorithm iterations.", 50, 10, 100)
gen.add("threshold", int_t, 0, "Cost threshold is used to limit the smoothing area.", 50, 1, 100)
gen.add("width_max_limit", double_t, 0, "Limit of the smoothing area.", 2, 0.5, 5)
gen.add("mean_angles_limit", double_t, 0, "When evaluating parameters a curved path is needed, which is characterised by mean value of absolute path angles in degrees.", 15.0, 5.0, 20.0)
gen.add("std_angles_limit", double_t, 0, "When evaluating parameters a curved path is needed, which is characterised by standard deviation of absolute path angles in degrees.", 20.0, 5.0, 25.0)
gen.add("path_length_limit", int_t, 0, "Minimal length of path needed for evaluation.", 15, 10, 25)

exit(gen.generate(PACKAGE, "path_smoothing", "Params"))
